CMakeLists.txt

```diff
@@ -152,6 +153,7 @@ if(HAVE_MKSTEMPS)
     set(CMAKE_C_FLAGS         "${CMAKE_C_FLAGS} -DHAVE_MKSTEMPS")
 endif(HAVE_MKSTEMPS)
 CHECK_STRUCT_HAS_MEMBER("struct stat" st_mtim "sys/stat.h" HAVE_STAT_ST_MTIM)
+CHECK_C_SOURCE_COMPILES("#define _GNU_SOURCE\n#include <fcntl.h>\nint main(int argc, char* argv[]) {return F_OFD_SETLKW;}" HAVE_LINUX_OFD_LOCK)
```

Comment:
I thought you would use simple `CHECK_SYMBOL_EXISTS` on `F_OFD_SETLKW`. This is a legacy way of doing things from `autotools`, inefficient.

Reply:
I thought CHECK_SYMBOL_EXISTS will not detect macro definition, but now I see from the cmake doc page:
"""
If the header files define the symbol as a macro it is considered available and assumed to work
"""

so yes will fix it! :)